tests/commands/test_new.py

```diff
@@ -36,9 +49,27 @@ def test_new(repo: Repo, directory: str) -> None:
     # verify correct output
     assert "The following will be created:" in ret.stdout
     assert "Create assets? (y/n) " in ret.stdout
+    assert file in ret.stdout
+    assert not ret.stderr
+    assert ret.returncode == 0
+    assert Path(file).exists()
+
+    # verify that the new asset exists and the repository is in a clean state
+    assert len(repo.assets) == 1
+    repo.fsck()
 
-    # double-casting needed to remove ./ in beginning
-    assert str(Path(file)) in ret.stdout
+
+def test_new_top_level(repo: Repo) -> None:
```

Comment:
This looks like a single-version case of `test_new()`. It should be possible to cover this special-case by adding `'./'` or `''` to the list of `directories` to test.

Reply:
I find adding either one of these to lead to different problems, which are just solvable when doing much more casting between strings and Paths. If that is prefered, I can change this, but it will effect all tests using `directories`, and I am not sure if it actually looks better.

Reply:
That surprises me. Can you share the specific error you get?

Reply:
Yes, but it is just an Assertion-error, because "./" gets removed for onyo output, so it needs the double-casting `str(Path(asset)) in ret.stdout` again, which I hate.

For `''` I forgot the error, but it is caused by e.g. `file = f'{directory}/laptop_apple_macbookpro.0'` being then just `/laptop_apple_macbookpro.0`. Again, this can be solved by double-casting in that line, but I do not like either of these solutions.

Though in general I am all for adding it to the directories, which is why I had it like that for the longest time, and just removed the double casting in this PR, so if you know any solution that works nicely without casting, I would be super happy! :)

Reply:
[Path.relative_to()](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.relative_to) might be of help.

Reply:
I think it is not, because it still needs to use a Path, that then gets casted into a string again for the output check.

The general problem is that we want to use some Paths and some strings in almost all test cases, so in my opinion this generally does never really work with just one variable without some casting.

The only alternative way I could see, is like in the beginning of set: https://github.com/psyinfra/onyo/blob/main/tests/commands/test_set.py
Defining assets/directories already and then building the carthesian product, so that pre-defined strings are given into the function, and then just casting it to Path() when needed. Though I am not sure, if we want to do that in general, because I feel like this creates a lot of unneeded test-cases.

In any case I think the general Path/Test handling should be discussed at some point and a uniform way of doing might be looked for. I remember you wanting to define a matrix for paths that get tested for all commands at some point, and I think in such an effort this should be discussed.